safegraph_patterns/delphi_safegraph_patterns/process.py

```diff
@@ -0,0 +1,199 @@
+        mapping info from naics_code to safegraph_brand_id
+
+    Returns
+    -------
+    dict{key(str): value(pd.DataFrame)}
+        Keys are metrics
+        Values are Dataframes with columns: timestamp, Zip Codes, {metric}_num.
+    """
+    result_dfs = {}
+    for metric, naics_code in zip(metric_names, naics_codes):
+        # Bar or restaurants
+        selected_brand_id = brand_df.loc[
+                brand_df["naics_code"] == naics_code, "safegraph_brand_id"]
+        filtered_df = df[df["safegraph_brand_ids"].isin(selected_brand_id)]
+        metric_count_name = "_".join([metric, "num"])
+
+        # Create wide df of visit counts for each day of week, along with zip and date info
+        # The counts end up as object by default, so explicitly cast as ints
+        visits_df = pd.DataFrame(
+                filtered_df["visits_by_day"].str[1:-1].str.split(",").tolist(),
+                index=filtered_df.index).astype("int")
+        visits_df["zip"] = filtered_df["postal_code"]
+        # Just keep date only
+        visits_df["start_date"] = pd.to_datetime(filtered_df["date_range_start"], utc=True)
+        visits_df["start_date"] = visits_df["start_date"].dt.normalize().dt.tz_localize(None)
+
+        # Turn df into long format and calculate actual dates from start_date and day of week
+        visits_long = visits_df.melt(
+                id_vars=["zip", "start_date"],
+                var_name="day_of_week",
+                value_name=metric_count_name)
+        visits_long[metric_count_name] = visits_long[metric_count_name].astype("int")
+        day_offsets = pd.to_timedelta(visits_long["day_of_week"], "d")
+        visits_long["timestamp"] = visits_long["start_date"] + day_offsets
+        visits_long.drop(["start_date", "day_of_week"], axis=1, inplace=True)
+
+        # Aggregate sum across same timestamps and zips
+        result_dfs[metric] = visits_long.groupby(
+                ["timestamp", "zip"]).sum().reset_index()
+
+    # Can have ~30k visits in restaurants missed in those zips in one week.
+    # only ~200 visits missed for bars.
+    return result_dfs
+
+def aggregate(df, metric, geo_res, map_df):
+    """
+    Aggregate signals to appropriate resolution.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        Zip Code-level data with prepared metrics (output of
+        construct_metrics().
+    metric: str
+        Name of metric to be exported.
+    geo_resolution: str
+        One of ('county', 'hrr, 'msa', 'state')
+    map_df: pd.DataFrame
+        population information and mapping info among different geo levels
+
+    Returns
+    -------
+    pd.DataFrame:
+        DataFrame with one row per geo_id, with columns for the individual
+        signals.
+    """
+    df = df.copy()
+    metric_count_name = "_".join([metric, "num"])
+    metric_prop_name = "_".join([metric, "prop"])
+
+    ############################ Use GeoMapper in utils #####################
+#    gmpr = GeoMapper()
```

Comment:
Please uncomment this block, test (I think the methods in gmpr used here are no longer available), and repush.  Ping me on Slack once this is ready and I will take another look ASAP :)

Reply:
I merged the new geo utils from main to this branch and it should work now. The unit tests have been updated